app_grim_reefer/src/buzzer.c

```diff
@@ -0,0 +1,65 @@
+#include "buzzer.h"
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(buzzer);
+
+K_MSGQ_DEFINE(buzzer_msgq, sizeof(enum buzzer_cond), 10, 1);
+
+void buzzer_tell(enum buzzer_cond cond) { k_msgq_put(&buzzer_msgq, &cond, K_MSEC(10)); }
+
+#define BUZZER_THREAD_STACK_SIZE 500
+#define BUZZER_THREAD_PRIORITY   10
+
+// 2 second beepcode, 2 second pause
+const uint16_t beepcodes[] = {
+    [buzzer_cond_noflash] = 0b1010101010101010,
+    [buzzer_cond_low_battery] = 0b1110001110001110,
+    [buzzer_cond_missing_sensors] = 0b1100000011000000,
+    [buzzer_cond_landed] = 0b1111111100000000,
+    [buzzer_cond_ok] = 0b0000000000000000,
+};
+
+extern void buzzer_entry_point(void *buzzer_gpio, void *, void *) {
```

Comment:
Don't think you need the extern here. You can also see about removing the extra 2 void * args

Reply:
Good point on extern. Removing the 2 extra void*s means the functions the wrong pointer type for starting a thread with and it gives a warning.

Reply:
Try making it static maybe? Think I've started threads in the past that I could just have as void args